File: Submissions/539_Minimum_Time_Difference.cpp
class Solution {
public:
    int findMinDifference(vector<string>& timePoints) {
        sort(timePoints.begin(), timePoints.end());
        
        int n = timePoints.size();
        int prev = getMinute(timePoints[n-1]);
        int curr = getMinute(timePoints[0]);
        int result = std::min(prev-curr, curr+24*60-prev);
        prev = curr;
        for(int i=1; i<n; ++i) {
            curr = getMinute(timePoints[i]);
            result = std::min(result, curr-prev);
            prev = curr;
        }
        
        return result;
    }
    
    int getMinute(const string& time){
        return (time[0] - '0') * 10 * 60 + (time[1] - '0') * 60 + (time[3] - '0') * 10 + (time[4] -'0'); 
    }
};

File: Kotlin/539_Minimum_Time_Difference.kt
class Solution {
    fun findMinDifference(timePoints: List<String>): Int {
        val sorted = timePoints.sorted() 
        
        val n = timePoints.size
        var prev = getMinute(sorted[n-1])
        var curr = getMinute(sorted[0])
        var result = minOf(prev - curr, curr + 24 * 60 - prev)
        prev = curr
        for(i in 1 until n) {
            curr = getMinute(sorted[i])
            result = minOf(result, curr-prev)
            prev = curr
        }
        
        return result
    }
    
    fun getMinute(time: String): Int {
        return (time[0] - '0') * 10 * 60 + (time[1] - '0') * 60 + (time[3] - '0') * 10 + (time[4] -'0')
    }
}